x-pack/plugins/apm/public/components/shared/charts/comparison_spark_plot/index.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React from 'react';
+import { EuiIcon } from '@elastic/eui';
+import {
+  AreaSeries,
+  Chart,
+  CurveType,
+  LineSeries,
+  ScaleType,
+  Settings,
+} from '@elastic/charts';
+import { merge } from 'lodash';
+import { Coordinate } from '../../../../../typings/timeseries';
+import { useChartTheme } from '../../../../../../observability/public';
+import { px, unit } from '../../../../style/variables';
+import { useTheme } from '../../../../hooks/use_theme';
+import { getComparisonChartTheme } from '../../time_comparison/get_time_range_comparison';
+import { Color } from '../spark_plot';
+
+export function ComparisonSparkPlot({
+  color,
+  series,
+  valueLabel,
+  compact,
+  comparisonSeries = [],
+}: {
+  color: Color;
+  series?: Coordinate[] | null;
+  valueLabel: React.ReactNode;
+  compact?: boolean;
+  comparisonSeries?: Coordinate[];
+}) {
+  const theme = useTheme();
+  const defaultChartTheme = useChartTheme();
+  const comparisonChartTheme = getComparisonChartTheme(theme);
+
+  const sparkplotChartTheme = merge({}, defaultChartTheme, {
+    chartMargins: { left: 0, right: 0, top: 0, bottom: 0 },
+    lineSeriesStyle: {
+      point: { opacity: 0 },
+    },
+    ...comparisonChartTheme,
+  });
+
+  const colorValue = theme.eui[color];
+
+  const chartSize = {
+    height: px(24),
+    width: compact ? px(unit * 3) : px(unit * 4),
+  };
+
+  return (
+    <EuiFlexGroup gutterSize="m" responsive={false}>
+      <EuiFlexItem grow={false}>
+        {!series || series.every((point) => point.y === null) ? (
+          <EuiIcon type="visLine" color="subdued" style={chartSize} />
+        ) : (
```

Comment:
nit: I wouldn't mind an isEmptyTimeseries() function here but if you feel that's overkill that's fine too.